guides/source/active_record_postgresql.md

```diff
@@ -524,6 +525,40 @@ user = User.create(name: 'John')
 User.last.name_upcased # => "JOHN"
 ```
 
+Deferrable Foreign Keys
+-----------------------
+
+* [foreign key table constraints](https://www.postgresql.org/docs/current/sql-set-constraints.html)
+
+By default, table constraints in PostgreSQL are checked immediately after each statement. It intenionally does not allow creating records where the referenced record is not yet in the referenced table. It is possible to run this integrity check later on when the transactions is committed by adding `DEFERRABLE` to the foreign key definition though. To defer all checks by default it can be set to `DEFERRABLE INITIALLY DEFERRED`. Rails exposes this PostgreSQL feature by adding the `:deferrable` key to the `foreign_key` options in the `add_reference` and `add_foreign_key` methods.
```

Comment:
A small typo: `intenionally` should be `intentionally`

Reply:
Fixed